cli/udf-wrapper/index.ts

```diff
@@ -0,0 +1,308 @@
+/// @ts-expect-error set individually for each UDF
+import udf from '${UDF_MAIN_FILE_PATH}'
+// import { IncomingMessage, ServerResponse, createServer } from 'http'
+// import { Readable } from 'stream'
+import { KVNamespace } from '@miniflare/kv'
+import { MemoryStorage } from '@miniflare/storage-memory'
+
+interface LogEntry {
+  loggedAt: number
+  level: string
+  message: string
+}
+
+interface FetchRequest {
+  loggedAt: number
+  url: string
+  method: string
+  statusCode: number
+  duration: number
+  contentType?: string
+  body: string | null
+}
+
+interface UdfRequestPayload {
+  info: unknown
+  parent: unknown
+  context: { kv?: unknown }
+  args: unknown
+}
+
+// type NodeResponse = ServerResponse<IncomingMessage> & { req: IncomingMessage }
+
+// type Headers = Record<string, string | string[]>
+
+enum HttpMethod {
+  Get = 'GET',
+  Post = 'POST',
+}
+
+enum HttpStatus {
+  NotFound = 404,
+  MethodNotAllowed = 405,
+}
+
+enum LogLevel {
+  Debug = 'debug',
+  Error = 'error',
+  Info = 'info',
+  Warn = 'warn',
+}
+
+enum MimeType {
+  ApplicationJson = 'application/json',
+  TextHtml = 'text/html',
+  TextPlain = 'text/plain',
+}
+
+enum Route {
+  Health = '/health',
+  Invoke = '/invoke',
+}
+
+enum ErrorType {
+  GraphQL = 'GraphQLError',
+}
+
+enum Header {
+  ContentType = 'content-type',
+}
+
+// enum Duplex {
+//   Half = 'half',
+// }
+
+// enum StreamEvent {
+//   Data = 'data',
+//   End = 'end',
+// }
+
+const PORT = 0
+// const HOST = '127.0.0.1'
+// const DUMMY_HOST = 'https://grafbase-cli'
+const MIME_PROPERTY_SEPARATOR = ';'
+
+const originalFetch = globalThis.fetch
+
+let logEntries: Array<LogEntry> = []
+let fetchRequests: Array<FetchRequest> = []
+
+// // Node.js:
```

Comment:
Is this helpful to keep? will likely be out of sync quite soon

Reply:
It's likely useful if we want to quickly switch back to node since the integration there isn't straightforward, but after it makes it into git history we can probably remove it